service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -136,19 +140,48 @@ private boolean checkSchemaIdSet() {
   }
 
   private boolean registerSchemas() {
+    List<GetSchemaResponse> existSchemas = srClient.getSchemas(microservice.getServiceId());
     for (Entry<String, String> entry : microservice.getSchemaMap().entrySet()) {
       String schemaId = entry.getKey();
       String content = entry.getValue();
-
-      boolean exists = srClient.isSchemaExist(microservice.getServiceId(), schemaId);
-      LOGGER.info("schemaId {} exists {}", schemaId, exists);
+      GetSchemaResponse existSchema = extractSchema(schemaId, existSchemas);
+      boolean exists = existSchema == null ? false : true;
+      LOGGER.info("schemaId [{}] exists {}", schemaId, exists);
       if (!exists) {
         if (!srClient.registerSchema(microservice.getServiceId(), schemaId, content)) {
           return false;
         }
+      } else {
+        String curSchemaSumary = existSchema.getSummary();
+        String schemaSummary = Hashing.md5().newHasher().putString(content, Charsets.UTF_8).hash().toString();
+        if (!schemaSummary.equals(curSchemaSumary)) {
+          if (microservice.getInstance().getEnvironment().startsWith("dev")) {
```

Comment:
startsWith?
why not equals?

Reply:
full name is too long, servicecenter use dev/prod, so sdk shuold match short name

Reply:
i don't know what's your mean.